include/sparrow/arrow_array_schema_proxy.hpp

```diff
@@ -374,14 +377,68 @@ namespace sparrow
          */
         [[nodiscard]] SPARROW_API arrow_proxy view() const;
 
+        /**
+         * Check if the proxy owns the `ArrowArray` and not only a pointer to it.
+         */
         [[nodiscard]] SPARROW_API bool owns_array() const;
+
+        /**
+         * Extract the `ArrowArray` from the proxy. The `ArrowArray` is not released when the proxy is
+         * destroyed.
+         * @exception `arrow_proxy_exception` If the `ArrowArray` was not created with sparrow.
+         * @details The schema flags can be updated by adding sparrow::ArrowFlag::NULLABLE, if null_count is
+         * greater than 0.
+         * @return The array.
+         */
         [[nodiscard]] SPARROW_API ArrowArray extract_array();
+
+        /**
+         * Get the reference to the `ArrowArray`. The `ArrowArray` is released when the proxy is destroyed.
```

Comment:
I would do something symmetrical to the extract_array documentation suggestion: ` Get a reference to the ` ArrowArray`  of the proxy. The proxy is still reponsible for releasing it, and the reference returned from this method should not outlive the proxy.

Reply:
ok